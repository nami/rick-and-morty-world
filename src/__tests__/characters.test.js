import React from "react";
import axios from 'axios';
import { create } from "react-test-renderer";
import Characters from '../components/characters'

var residents = []

let getResidents = () => {
		let url = 'https://rickandmortyapi.com/api/location/'
		axios.get(url)
    .then((res) => {
    	residents = res['data']['results']['residents']
    })
    .catch((error) =>{
      console.log(error)
    })
  return residents
}

describe("Characters component", () => {
  test("Matches the snapshot", () => {
    const characters = create(<Characters residents={getResidents()}/>);
    expect(characters.toJSON()).toMatchSnapshot();
  });
});