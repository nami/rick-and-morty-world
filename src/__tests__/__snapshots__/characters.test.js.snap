// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component Matches the snapshot 1`] = `
<div
  className="characters"
/>
`;
